api/machinemanager/machinemanagernew_test.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package machinemanager_test
+
+import (
+	"github.com/golang/mock/gomock"
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/api/base/mocks"
+	"github.com/juju/juju/api/machinemanager"
+	"github.com/juju/juju/apiserver/params"
+	jujutesting "github.com/juju/juju/testing"
+)
+
+var _ = gc.Suite(&NewMachineManagerSuite{})
+
+type NewMachineManagerSuite struct {
```

Comment:
What is the rationale behind the new file/suite here?

Reply:
The existing `machinemanager_test.go` uses a home grown mocking mechanism. I wanted to establish the `new` suffixed file to have a place to start systematically moving those tests to use gomock. There are two benefits to this:

1) We can work piecemeal
2) We don't have to mix two mocking styles (in attempt to preserve one file) when transitioning between mocking styles

The plan is to start moving those old style tests and when finished delete the old file and `mv` the new file.

Reply:
Understood - strangulation.
Maybe a comment at the top along these lines?

Reply:
Going to place my comment above in the code at the top of the file.